### PR TITLE
Initial rollback documentation.

### PR DESCRIPTION
**Pull request by will-in-wi**
This is an initial draft of rollback documentation. I expect this to go through a couple iterations.

For discussion on https://github.com/capistrano/documentation/issues/124.

### CODE REVIEW

File: documentation/getting-started/rollbacks/index.markdown [hunk 1]
```diff
@@ -3,4 +3,18 @@ title: Rollbacks
 layout: default
 ---
 
-**Missing Content**
+When a deployment is run, Capistrano executes one task at a time on all servers and waits for that task to be done before moving on to the next one. If a task fails on one or more servers, the errors are printed, Capistrano waits for all servers to either return an error or succeed, and then stops execution of any further tasks.
```
Comment (mattbrictson) [inline documentation/getting-started/rollbacks/index.markdown]: This is not necessarily true. Generally speaking, Capistrano runs a task on all servers in parallel. It then waits for the first server to complete (or fail), then the second, and so on. If a failure is detected, Capistrano exits without waiting for the remaining servers to finish.

For example, if you have servers 1, 2, and 3, and 3 fails, Capistrano won't "know" until 1 and 2 are done. In this case all servers execute before Capistrano exits.

If 1 fails, Capistrano exits right away before checking the status of 2 and 3. If 2 and 3 haven't finished yet, the SSH connection is simply terminated (AFAIK).

Not sure how to summarize that simply in the doc, but there you go. :)

**Comment by mattbrictson** [comment]
Despite using Capistrano almost daily for many years, I don't think I have ever tried the rollback feature. So I'm not sure I'm the best person to review this PR, but I'll try to help as much as I can.

Based on looking at the code (which I assume you are doing as well), it seems that rollback is identical to deploy, except the part where Capistrano determines what release directory to use. In a deploy, a new release directory is created and the latest code is checkout from SCM into it. This becomes "current". In rollback, the release immediately prior to the current one is elected "current". Then, this new "current" is deployed.

(There's also the extra step of tar-ing up and archiving, which you mentioned in your doc.)

It might also be worth pointing out that depending on your app and frameworks you use, there may be additional steps needed to perform a proper rollback. Capistrano is a general purpose tool and cannot do this for you. For example, you may want to roll back a database migration. Capistrano plugins may provide this functionality, or you may have to write your own. The capistrano-rails plugin, for example, will rollback assets, but it does not rollback database migrations, IIRC.

Sorry these comments are all over the place; I am just coming to grips with the rollback feature myself.

**Comment by leehambley** [comment]
I'm in the same boat as @mattbrictson - I don't believe in "rollback" and it's _so_ dependent on languages, frameworks, plugins, etc. I've always found it to be more reliable to _re-deploy_ a last-known good version, rather than relying on seldom used code path in a time of crisis.

That said, I'd rather make these concerns part of the rollback docs, than start discussing pulling the feature. It's rare that a "rollback" is so urgent that you can't afford the time to deploy again in my experience. I'll get back to you **_today**_. This is important, and your contribution is in a thankless area of doc-writing.

**Comment by will-in-wi** [comment]
I agree, I always end up reverting the branch and deploying again. However, that's partially been since I didn't know enough about rollback. So, I want to learn a little bit and get these docs up to snuff. Thanks for the thoughts! I'll try to find some time today to do further edits based on both of your comments.

**Comment by mattbrictson** [comment]
Great discussion. Can we drop this into the docs somewhere? :grin: 

> Correctly rolling back a release is a complex process that depends on the specifics of your application and the Capistrano plugins you've assembled. **Be proactive and test your rollback procedure before trying it for the first time in a time of crisis.**

**Comment by will-in-wi** [comment]
I've cleaned up a little and clarified per the comments you guys have made.

If you and/or others could re-read and give further thoughts (grammar and clarity nitpicking is welcome), I'd appreciate it. Or just clarify what remains before merging.

Thanks for your help on this!

**Comment by leehambley** [comment]
@mattbrictson I'm happy to merge this, any last comments?

**Comment by mattbrictson** [comment]
Fantastic. LGTM :sparkles:

**Pull request merged**